go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -590,3 +661,106 @@ func getClusterSubnetsExclusion() string {
 	}
 	return exclusion
 }
+
+// AddNodeDNSInfo is called when a NodeDNSInfo object is created, resposible for creating initial data
+// for dnsNames resolved by nodes
+func (efDNS *egressfirewallDNSInformation) AddNodeDNSInfo(nodeDNSInfo *nodednsinfoapi.NodeDNSInfo) error {
+	klog.Infof("Adding NodeDNSInfo %s to the cluster", nodeDNSInfo.Name)
+
+	efDNS.Lock()
+	defer efDNS.Unlock()
+	for dnsName, dnsEntries := range nodeDNSInfo.Spec.DNSEntries {
+		var ipNet []net.IP
+		for _, ip := range dnsEntries.IPAddresses {
+			if efDNS.dnsNameInfo[dnsName].ipNodes[ip] == nil {
```

Comment:
Are we guaranteed that `efDNS.dnsNameInfo[dnsName]` exists and won't panic here in any imaginable case?

Reply:
Yes, `efDNS.dnsNameInfo[dnsName]` is created by addEgressFirewallRules() which creates it before signaling to the nodes to resolve the dns